RELEASE_CHECKLIST.md

```diff
@@ -4,16 +4,19 @@
 
 This document describes the checklist to publish a release via GitHub workflow.
 
-NOTE: Make sure the dependencies in `go.mod` file are expected by the release. For example, if there are dependencies on certain version of notation library (notation-go or notation-core-go) or ORAS library (oras-go), make sure that version of library is released first, and the version number is updated accordingly in `go.mod` file.  After updating go.mod file, run `go mod tidy` to ensure the go.sum file is also updated with any potential changes.
-
 ## Release Process
 
+1. Check if there are any security vulnerabilities fixed and security advisories published before a release. Security advisories should be linked on the release notes.
 1. Determine a [SemVer2](https://semver.org/)-valid version prefixed with the letter `v` for release. For example, `version="v1.0.0-alpha.1"`.
-1. Bump up the `Version` in [internal/version/version.go](internal/version/version.go#L5) and open a PR for the changes.
-1. Wait for the PR merge.
-1. Be on the main branch connected to the actual repository (not a fork) and `git pull`.  Ensure `git log -1` shows the latest commit on the main branch.
-1. Create a tag `git tag -am $version $version`
-1. `git tag` and ensure the name in the list added looks correct, then push the tag directly to the repository by `git push --follow-tags`.
+1. If there are code changes in notation-go or notation-core-go library, follow the two steps above and cut the release for the notation library. Then update the dependency versions in notation [go.mod](go.mod), [go.sum](go.sum), [test/e2e/go.mod](test/e2e/go.mod), and [test/e2e/go.sum](test/e2e/go.sum). Run `go mod tidy` to check if the `go.sum` file is also updated with any potential changes. Update the `Version` in [internal/version/version.go](internal/version/version.go#L5) in the forked repository. 

```

Comment:
"Update the `Version` in [internal/version/version.go](internal/version/version.go#L5) in the forked repository." <- This sentence should be removed.